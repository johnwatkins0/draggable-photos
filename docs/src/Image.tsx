import * as React from 'react';

import DraggableSplitscreen from '../../src/draggable-splitscreen';

const LeftSide = () => (
  <img
    src="/react-draggable-splitscreen/docs/assets/left-side.png"
    alt="Left Side"
    style={{ width: '100%', height: 'auto' }}
  />
);

const RightSide = () => (
  <img
    src="/react-draggable-splitscreen/docs/assets/right-side.png"
    alt="Right Side"
    style={{ width: '100%', height: 'auto' }}
  />
);

const Image = () => <DraggableSplitscreen leftSide={<LeftSide />} rightSide={<RightSide />} />;

export default Image;
