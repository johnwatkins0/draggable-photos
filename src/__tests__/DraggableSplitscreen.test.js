import React from 'react';
import renderer from 'react-test-renderer';

import Image from '../../example/src/Image';
import Text from '../../example/src/Text';
import Video from '../../example/src/Video';

describe('DraggableSplitscreen', () => {
  it('renders an image-based element.', () => {
    const tree = renderer.create(<Image />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders a text-based element', () => {
    const tree = renderer.create(<Text />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders a video-based element', () => {
    const tree = renderer.create(<Video />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

