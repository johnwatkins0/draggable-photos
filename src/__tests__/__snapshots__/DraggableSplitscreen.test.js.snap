// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableSplitscreen renders a text-based element 1`] = `
<div
  className="sc-bdVaJa cQtrjW"
>
  <div>
    <div
      style={
        Object {
          "backgroundColor": "blue",
          "padding": "6rem",
        }
      }
    >
      <h2>
        Left side
      </h2>
    </div>
  </div>
  <div
    className="sc-bwzfXH gbnVeh"
  >
    <div
      style={
        Object {
          "backgroundColor": "red",
          "padding": "6rem",
          "textAlign": "right",
        }
      }
    >
      <h2>
        Right side
      </h2>
    </div>
  </div>
  <div
    className="react-draggable sc-htpNat bhkyHK"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "touchAction": "none",
        "transform": "translate(0px,0px)",
      }
    }
    transform={null}
  />
</div>
`;

exports[`DraggableSplitscreen renders a video-based element 1`] = `
<div
  className="sc-bdVaJa cQtrjW"
>
  <div>
    <video
      autoPlay={true}
      loop={true}
      muted={true}
      src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"
      style={
        Object {
          "height": "auto",
          "objectFit": "inherit",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH gbnVeh"
  >
    <video
      autoPlay={true}
      loop={true}
      muted={true}
      src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4"
      style={
        Object {
          "height": "auto",
          "objectFit": "inherit",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="react-draggable sc-htpNat bhkyHK"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "touchAction": "none",
        "transform": "translate(0px,0px)",
      }
    }
    transform={null}
  />
</div>
`;

exports[`DraggableSplitscreen renders an image-based element. 1`] = `
<div
  className="sc-bdVaJa cQtrjW"
>
  <div>
    <img
      alt="Left Side"
      src="/react-draggable-splitscreen/example/assets/left-side.png"
      style={
        Object {
          "height": "auto",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH gbnVeh"
  >
    <img
      alt="Right Side"
      src="/react-draggable-splitscreen/example/assets/right-side.png"
      style={
        Object {
          "height": "auto",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="react-draggable sc-htpNat bhkyHK"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "touchAction": "none",
        "transform": "translate(0px,0px)",
      }
    }
    transform={null}
  />
</div>
`;
